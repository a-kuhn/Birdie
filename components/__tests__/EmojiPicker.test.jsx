import React from "react";
import { Text } from "react-native";
import { render, fireEvent } from "@testing-library/react-native";
import EmojiPicker from "../EmojiPicker";

jest.mock("@expo/vector-icons/MaterialIcons", () => "MaterialIcons");

describe("EmojiPicker", () => {
  it("renders correctly when visible", () => {
    const { getByText, getByTestId, debug } = render(
      <EmojiPicker isVisible={true} onClose={() => {}}>
        <Text testID="child">Child content</Text>
      </EmojiPicker>
    );
    expect(getByText("Choose a sticker")).toBeTruthy();
    expect(getByTestId("child")).toBeTruthy();
  });

  it("does not render when not visible", () => {
    const { queryByText, queryByTestId } = render(
      <EmojiPicker isVisible={false} onClose={() => {}}>
        <Text testID="child">Child content</Text>
      </EmojiPicker>
    );
    expect(queryByText("Choose a sticker")).toBeNull();
    expect(queryByTestId("child")).toBeNull();
  });

  it('calls onClose when close button is pressed', () => {
    const onCloseMock = jest.fn();
    const { getByRole } = render(
      <EmojiPicker isVisible={true} onClose={onCloseMock}>
        <Text>Child content</Text>
      </EmojiPicker>
    );
    fireEvent.press(getByRole("button"));
    expect(onCloseMock).toHaveBeenCalled();
  });

  it("matches snapshot when visible", () => {
    const { toJSON } = render(
      <EmojiPicker isVisible={true} onClose={() => {}}>
        <Text>Child content</Text>
      </EmojiPicker>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("matches snapshot when not visible", () => {
    const { toJSON } = render(
      <EmojiPicker isVisible={false} onClose={() => {}}>
        <Text>Child content</Text>
      </EmojiPicker>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});