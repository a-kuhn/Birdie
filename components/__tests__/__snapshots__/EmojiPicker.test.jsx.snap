// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiPicker matches snapshot when not visible 1`] = `null`;

exports[`EmojiPicker matches snapshot when visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "backgroundColor": "#25292e",
        "borderTopLeftRadius": 18,
        "borderTopRightRadius": 18,
        "bottom": 0,
        "height": "25%",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#464C55",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "flexDirection": "row",
          "height": "16%",
          "justifyContent": "space-between",
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
          }
        }
        testID="emoji-picker-text"
      >
        Choose a sticker
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <MaterialIcons
          color="#fff"
          name="close"
          size={22}
        />
      </View>
    </View>
    <Text>
      Child content
    </Text>
  </View>
</Modal>
`;
